contrib/docker/ubuntu/noble/Dockerfile

```diff
@@ -0,0 +1,234 @@
+#######################
+#
+#  Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+#  agreements.  See the NOTICE file distributed with this work for additional information regarding
+#  copyright ownership.  The ASF licenses this file to you under the Apache License, Version 2.0
+#  (the "License"); you may not use this file except in compliance with the License.  You may obtain
+#  a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing permissions and limitations under
+#  the License.
+#
+#######################
+FROM ubuntu:noble AS build
+
+ARG LLVM_VERSION=18
+ARG BASE=/opt
+ARG ATS_VERSION=10.0.0
+
+RUN apt update \
+ && apt upgrade --yes \
+ && apt install --no-install-recommends --yes \
+    ca-certificates \
+    clang-${LLVM_VERSION} \
+    libc++-${LLVM_VERSION}-dev \
+    cmake \
+    ninja-build \
+    libc-ares-dev \
+    libsystemd-dev \
+    libev-dev \
+    libevent-dev \
+    zlib1g-dev \
+    cargo \
+    wget \
+    git \
+    libtool \
+    make \
+    pkg-config \
+# ATS deps
+    libxml2-dev \
+    libjemalloc-dev \
+    libhwloc-dev \
+    libfmt-dev \
+    libpcre2-dev \
+    libpcre3-dev \
+    hwloc \
+    libbrotli-dev \
+    luajit \
+    libcap-dev \
+    libmagick++-dev \
+    libmaxminddb-dev \
+    libcjose-dev \
+    libcjose0 \
+    libjansson-dev \
+ # needed to appease nghttp2 but unused
+    libssl-dev \
+ && apt clean --yes
+
+RUN mkdir -p ${BASE} && chmod a+rX ${BASE}
+
+RUN if [ `uname -m` = "arm64" -o `uname -m` = "aarch64" ]; then echo "arm64" > /arch; else echo "amd64" > /arch; fi \
+  && wget -qO- https://go.dev/dl/go1.21.6.linux-$(cat /arch).tar.gz | tar -C ${BASE} -xzf -
+
+ENV CC=clang-${LLVM_VERSION}
+ENV CXX=clang++-${LLVM_VERSION}
+ENV GO_BINARY_PATH=${BASE}/go/bin/go
+
+RUN git clone https://boringssl.googlesource.com/boringssl \
+ && cd boringssl \
+ && git checkout 45b2464158379f48cec6e35a1ef503ddea1511a6 \
```

Comment:
For these libraries, can we copy down the build_h3_tools.sh script and run that instead? It would be nice to avoid two implementations of the invocations of these builds if we can.

Reply:
I don't think it would work as-is, but maybe we could do that?

Reply:
It would be nice to update the script to work in Ubuntu if it's not too much trouble. That would be a win for this Dockerfile and others trying to use the script in Ubuntu.